x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/group_of_nodes.tsx

```diff
@@ -43,18 +43,19 @@ export const GroupOfNodes: React.FC<Props> = ({
     <GroupOfNodesContainer style={{ width }}>
       <GroupName group={group} onDrilldown={onDrilldown} isChild={isChild} options={options} />
       <Nodes>
-        {group.nodes.map((node) => (
-          <Node
-            key={`${node.pathId}:${node.name}`}
-            options={options}
-            squareSize={group.squareSize}
-            node={node}
-            formatter={formatter}
-            bounds={bounds}
-            nodeType={nodeType}
-            currentTime={currentTime}
-          />
-        ))}
+        {!isNaN(group.squareSize) &&
```

Comment:
Do we understand why this value is NaN? I like this change but I wonder if this is the best/right place to check for this?

Reply:
The problem is that the div that the Autosizer is watching is going to have a width of 0 during a few renders.  If the width is 0 the [applyWaffleMapLayout will return squareSize as NaN](https://github.com/elastic/kibana/blob/master/x-pack/plugins/infra/public/pages/metrics/inventory_view/lib/apply_wafflemap_layout.ts#L67).  One thing that might be nicer  which should work just the sam is to replace `!isNaN(group.squareSize)` with `group.width`.  If group.width is not zero, squareSize should never be NaN.  I think this is the best place for the condition because we still want to render the container node of Node which is `Nodes`.  There are some styles which would change how things look a bit while loading.

Reply:
OK sounds good. I think what you have is good for now.

Reply:
@jasonrhodes  Later we can dig deeper and see if we can do something better but this change is low risk and doesn't involve changing any structure around.

Reply:
Yeah, that makes sense. Thanks!